pkg/steps/artifacts.go

```diff
@@ -216,6 +216,27 @@ type PodClient interface {
 	podCmdExecutor
 }
 
+func waitForContainer(podClient PodClient, ns, name, containerName string) error {
+	logrus.Tracef("Waiting for container %s in pod %s in namespace %s", containerName, name, ns)
+	for {
+		pod, err := podClient.Pods(ns).Get(name, meta.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		for _, container := range pod.Status.ContainerStatuses {
+			if container.Name == containerName {
+				if container.State.Running != nil || container.State.Terminated != nil {
+					return nil
+				}
+				break
+			}
+		}
+
+		time.Sleep(time.Second)
```

Comment:
Can we use `wait.PollImmediate()`

Reply:
> Can we use `wait.PollImmediate()`

Pivoted to `PollImmediateInfinite` with 6c2d3168 -> ea588d07, because `copyArtifacts` does not take a `Context` or timeout duration or other limit.  If you want `PollImmediate`, did you have a preference around an internal, hard-coded timeout or plumbing through a context/timeout from a higher level?

Reply:
This should be quick and rarely fail, I don't think sending in the interruption context would be worth the implementation complexity (e.g. we still want to gather artifacts for `post` multi-stage steps even after an interruption). I would still add a reasonably long timeout just so it doesn't get stuck forever, unlikely as that may be.

Reply:
Pick a reasonably long timeout for me?

Reply:
The containers should start at roughly the same instant, so something like `5m`?